### PR TITLE
BUG: DataFrame.drop raising TypeError for string label with non-unique MultiIndex and no level provided

### LINKED ISSUE #36293: BUG: `DataFrame.drop` fails when there is a multiIndex without any level provided
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
>>> x = pd.DataFrame({"a": range(10), "b": range(10, 20), "d": ["a", "v"] * 5}, index=pd.MultiIndex(levels=[['lama', 'cow', 'falcon'],
...                              ['speed', 'weight', 'length']],
...                      codes=[[0, 0, 0, 1, 1, 1, 2, 2, 2, 1],
...                             [0, 1, 2, 0, 1, 2, 0, 1, 2, 1]]))
>>> x
               a   b  d
lama   speed   0  10  a
       weight  1  11  v
       length  2  12  a
cow    speed   3  13  v
       weight  4  14  a
       length  5  15  v
falcon speed   6  16  a
       weight  7  17  v
       length  8  18  a
cow    weight  9  19  v
>>> x.drop(index='cow')
Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/frame.py", line 4169, in drop
    errors=errors,
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/generic.py", line 3884, in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/generic.py", line 3933, in _drop_axis
    indexer = ~axis.isin(labels)
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/indexes/multi.py", line 3600, in isin
    values = MultiIndex.from_tuples(values, names=self.names)._values
  File "/Users/pgali/PycharmProjects/del/venv1/lib/python3.7/site-packages/pandas/core/indexes/multi.py", line 501, in from_tuples
    arrays = list(lib.tuples_to_object_array(tuples).T)
  File "pandas/_libs/lib.pyx", line 2471, in pandas._libs.lib.tuples_to_object_array
TypeError: Expected tuple, got str


```

#### Problem description

I'd expected `drop` to drop all the rows with `cow` as an index.

#### Expected Output

```python
>>> x.drop(index='cow')
               a   b  d
lama   speed   0  10  a
       weight  1  11  v
       length  2  12  a
falcon speed   6  16  a
       weight  7  17  v
       length  8  18  a
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 2a7d3326dee660824a8433ffd01065f8ac37f7d6
python           : 3.7.3.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.6.0
Version          : Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_US.UTF-8
pandas           : 1.1.2
numpy            : 1.17.3
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 49.1.0
Cython           : None
pytest           : None
hypothesis       : 5.29.0
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by phofl**
- [x] closes #36293
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

As far as I know, isin needs a level input when only a string is given for a MultiIndex. If this should not change, we have to set level=0 in this case.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -4173,6 +4173,14 @@ def _drop_axis(
 
         # Case for non-unique axis
         else:
+            if (
```
Comment (jreback) [inline pandas/core/generic.py:4176]: should this case only be an issue starting on L4193? e.g. level is None there

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 1 above)
Comment (phofl) [reply pandas/core/generic.py:4176]: Yep, this is correct. Question is what to do when level is None.

I think ``~axis.get_level_values(level).isin(labels)`` is the way to go, so I check before reaching the if else block if level is None to set it to 0 in this case

Comment (jreback) [reply pandas/core/generic.py:4176]: i think setting to level = 0 is fine, i would just move the check *inside* the block on L4193 as its not going to be true where it is if level is not None in the first place.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 1 above)
Comment (phofl) [reply pandas/core/generic.py]: This would work as an alternative

File: pandas/core/generic.py [hunk 2]
```diff
@@ -4183,11 +4183,16 @@ def _drop_axis(
                 if errors == "raise" and indexer.all():
                     raise KeyError(f"{labels} not found in axis")
             else:
-                indexer = ~axis.isin(labels)
-                # Check if label doesn't exist along axis
-                labels_missing = (axis.get_indexer_for(labels) == -1).any()
-                if errors == "raise" and labels_missing:
-                    raise KeyError(f"{labels} not found in axis")
+                if isinstance(axis, MultiIndex) and labels.dtype == "object":
```
Comment (jreback) [inline pandas/core/generic.py:4186]: make an elif here

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 2 above)
Comment (phofl) [reply pandas/core/generic.py]: Done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**